CC/src/DistanceComputationTools.cpp

```diff
@@ -2677,6 +2703,66 @@ int DistanceComputationTools::computeCloud2BoxEquation(GenericIndexedCloudPersis
 	return count;
 }
 
+int DistanceComputationTools::computeCloud2PolylineEquation(GenericIndexedCloudPersist* cloud, const Polyline* polyline, double* rms /*= nullptr*/)
+{
+	assert(cloud);
+	if (!cloud)
+	{
+		return -1;
+	}
+	unsigned count = cloud->size();
+	if (count == 0)
+	{
+		return -2;
+	}
+	if (!cloud->enableScalarField())
+	{
+		return -3;
+	}
+	ScalarType d = 0;
+	ScalarType dSumSq = 0;
+	for (unsigned i = 0; i < count; ++i)
+	{
+		ScalarType distSq = INFINITY;
```

Comment:
I wasn't sure what to use for infinity here

Reply:
We normally use 'NAN_VALUE' (and you can use std::isnan to test it)

Reply:
I would argue that using NAN_VALUE is probably not the best choice here as the later use
```C++
distSq = std::min(distSq, computePoint2LineSegmentDistSquared(p, start, end));
```
Would become more complicated than required.

would it be acceptable to add the following to CCConst.h

```C++
//! INF as a ScalarType value
/** \warning: handle with care! **/
constexpr ScalarType INF_VALUE = std::numeric_limits<ScalarType>::infinity();

```
and then set any INF_VALUE at the end to NAN_VALUE if required outside of the inner loop?

if not then I would suggest we set the value to the first loop iteration value like:

```C++
ScalarType distSq = computePoint2LineSegmentDistSquared(p, 
							polyline->getPoint(0), 
							polyline->getPoint(1));
```

@dgirardeau what are your thoughts, is INF_VALUE a risk in other areas I'm not considering? Am I missing something that wouldn't require extra logic in the inner loop?

Reply:
Oh if it's just a temporary value, I definitely prefer to init the distance the first time with the right value. The infinity value wouldn't be properly handled by the other components of CloudCompare (especially at display time, etc.).